core/drivers/crypto/hisilicon/sec_hash.c

```diff
@@ -0,0 +1,674 @@
+		/* LONG BD MIDDLE */
+		sqe->auth_mac_key |= (AI_GEN_IVIN_ADDR << SEC_AI_GEN_OFFSET_V3);
+		sqe->stream_scene.auth_pad = AUTHPAD_NOPAD;
+		sqe->a_ivin_addr = sqe->mac_addr;
+		ctx->iv_len = ctx->mac_len;
+	} else if (!ctx->has_next && ctx->iv_len) {
+		/* LONG BD END */
+		sqe->auth_mac_key |= (AI_GEN_IVIN_ADDR << SEC_AI_GEN_OFFSET_V3);
+		sqe->stream_scene.auth_pad = AUTHPAD_PAD;
+		sqe->a_ivin_addr = sqe->mac_addr;
+		total_bits = ctx->long_data_len * BYTE_BITS;
+		sqe->stream_scene.long_a_data_len = total_bits;
+		ctx->iv_len = 0;
+	} else {
+		/* SHORT BD */
+		ctx->iv_len = 0;
+	}
+}
+
+static enum hisi_drv_status sec_digest_fill_bd3_sqe(void *bd, void *msg)
+{
+	struct hisi_sec_bd3_sqe *sqe = (struct hisi_sec_bd3_sqe *)bd;
+	enum hisi_drv_status ret = HISI_QM_DRVCRYPT_NO_ERR;
+	struct hashctx *ctx = (struct hashctx *)msg;
+	uint32_t alg_type = 0;
+
+	sqe->bd_param = (BD_TYPE3 | (ctx->scene << SEC_SCENE_OFFSET_V3));
+	sqe->a_len_key = ctx->in_len;
+	sqe->auth_mac_key = AUTH_MAC_CALCULATE;
+	sqe->data_src_addr = ctx->in_dma;
+	sqe->mac_addr = ctx->out_dma;
+
+	if (ctx->mode == WCRYPTO_DIGEST_HMAC) {
+		ret = sec_digest_set_hmac_bd3_key(ctx, sqe);
+		if (ret)
+			return HISI_QM_DRVCRYPT_IN_EPARA;
+	}
+
+	sqe->auth_mac_key |= ((ctx->mac_len / SEC_ENCODE_BYTES) <<
+			      SEC_MAC_OFFSET_V3);
+	alg_type = sec_digest_get_alg_type(ctx->algo);
+	if (alg_type >= A_ALG_MAX) {
+		EMSG("Fail to get digest bd3 alg");
+		return HISI_QM_DRVCRYPT_IN_EPARA;
+	}
+	sqe->auth_mac_key |= (alg_type << SEC_AUTH_ALG_OFFSET_V3);
+	sec_digest_fill_long_bd3(ctx, sqe);
+
+	return HISI_QM_DRVCRYPT_NO_ERR;
+}
+
+static TEE_Result sec_digest_do_task(struct hisi_qp *qp, void *msg)
+{
+	TEE_Result ret = TEE_SUCCESS;
+
+	ret = hisi_qp_send(qp, msg);
+	if (ret) {
+		EMSG("Fail to send task, ret=%d", ret);
+		return TEE_ERROR_BAD_STATE;
+	}
+
+	ret = hisi_qp_recv_sync(qp, msg);
+	if (ret) {
+		EMSG("Recv task error, ret=%d", ret);
+		return TEE_ERROR_BAD_STATE;
+	}
+
+	return TEE_SUCCESS;
+}
+
+static enum hisi_drv_status sec_parse_digest_sqe(void *bd, void *msg __unused)
+{
+	struct hisi_sec_sqe *sqe = (struct hisi_sec_sqe *)bd;
+	uint16_t done;
+
+	done = SEC_GET_FIELD(sqe->type2.done_flag, SEC_DONE_MASK, 0);
+	if (done != SEC_HW_TASK_DONE || sqe->type2.error_type) {
+		EMSG("SEC BD2 fail! done=0x%x, etype=0x%x",
+		     done, sqe->type2.error_type);
+		return HISI_QM_DRVCRYPT_IN_EPARA;
+	}
+
+	return HISI_QM_DRVCRYPT_NO_ERR;
+}
+
+static enum hisi_drv_status sec_parse_digest_bd3_sqe(void *bd,
+						     void *msg __unused)
+{
+	struct hisi_sec_bd3_sqe *sqe = (struct hisi_sec_bd3_sqe *)bd;
+	uint16_t done = 0;
+
+	done = SEC_GET_FIELD(sqe->done_flag, SEC_DONE_MASK, 0);
+	if (done != SEC_HW_TASK_DONE || sqe->error_type) {
+		EMSG("SEC BD3 fail! done=0x%x, etype=0x%x",
+		     done, sqe->error_type);
+		return HISI_QM_DRVCRYPT_IN_EPARA;
+	}
+
+	return HISI_QM_DRVCRYPT_NO_ERR;
+}
+
+TEE_Result sec_digest_ctx_init(struct hashctx *hash_ctx,
```

Comment:
Global functions should use a prefix. Wouldn't `hisi_sec_` make sense in this file? Same for all the other globals in this file.